cmd/client.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author: Evan Spendlove (@evanSpendlove)
+//
+// Client.go creates the client for connecting to, and making requests to, the
+// gRPC server of Cloudprober.
+//
+// TODOs:
+// TODO(#38) Add support for secure connection to RPC server with credentials, if supported by Cloudprober.
+// TODO: Include details of these errors (addProbeFromConfig method) in the error message.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/golang/glog"
+	proberpb "github.com/google/cloudprober/prober/proto"
+	cpprobes "github.com/google/cloudprober/probes"
+	configpb "github.com/google/cloudprober/probes/proto"
+	"google.golang.org/grpc"
+)
+
+type CloudproberClient struct {
+	conn      *grpc.ClientConn           // Client connection for opening and closing the connection
+	client    proberpb.CloudproberClient // Cloudprober gRPC client retained between RPC calls
+	clientMux sync.Mutex                 // Mutex lock for making CloudproberClient safe for concurrent use
+}
+
+// NewClient creates, initialises and returns a new CloudproberClient.
+// This is the correct way to create a new instance of CloudproberClient.
+// Parameters:
+// - rpcServer: Pass the rpcServer address and port as a string formatted as: "hostname:port", e.g. "localhost:1234"
+// Returns:
+// - error:
+//	  - nil: No error occurred and the client was successfully initialised.
+//	  - gRPC Error: See https://godoc.org/google.golang.org/grpc/codes for error codes.
+//		-> See https://github.com/grpc/grpc-go/blob/d25c71b54334380ff1febd25d88064b36de44b3c/clientconn.go#L123
+func NewClient(rpcServer string) (*CloudproberClient, error) {
+	conn, err := grpc.Dial(rpcServer, grpc.WithInsecure())
+	if err != nil {
+		return nil, fmt.Errorf("GRPC_DIAL_ERROR: %v", err)
+	}
+	return &CloudproberClient{conn: conn, client: proberpb.NewCloudproberClient(conn)}, nil
+}
+
+// CloseConn is used for closing the client connection with the gRPC server.
+// Returns:
```

Comment:
Sync the comment with the signature (currently the function doesn't return anything).

Reply:
I have updated this comment so that is reflects the current function signature.